src/test/scala/mesosphere/marathon/integration/ResidentTaskIntegrationTest.scala

```diff
@@ -19,88 +21,189 @@ class ResidentTaskIntegrationTest
+    val app = F.create(F.residentApp(instances = 5, upgradeStrategy = UpgradeStrategy(0.5, 0.0)))
+
+    When("we change the config")
+    val newVersion = F.update(app.id, AppUpdate(cmd = Some("sleep 1234"))).toString
+    val all = F.allTasks(app.id)
+
+    Then("all 5 tasks are of the new version")
+    all.map(_.version).forall(_.contains(newVersion)) shouldBe true
+
+    And("exactly 5 instances are running")
+    all.count(_.launched) shouldBe 5
+
+    And("no extra task was created")
+    all.size shouldBe 5
+  }
+
+  /**
+    * FIXME (3043): implement the following tests. TASK_LOST can be induced when launching a task with permission:
+    *
+    * When a framework launches a task, “run_tasks” ACLs are checked to see if the framework
+    * (FrameworkInfo.principal) is authorized to run the task/executor as the given user. If not authorized,
+    * the launch is rejected and the framework gets a TASK_LOST.
+    *
+    * (From http://mesos.apache.org/documentation/latest/authorization/)
+    */
+
+  ignore("taskLostBehavior = RELAUNCH_AFTER_TIMEOUT, timeout = 10s") {
+    Given("A resident app with 1 instance")
+    When("The task is lost")
+    Then("The task is not relaunched within the timeout")
+    And("The task is relaunched with a new Id after the timeout")
   }
 
-  def waitForStatusUpdates(kinds: String*) = kinds.foreach { kind =>
-    waitForEventWith(Event.STATUS_UPDATE_EVENT, _.info("taskStatus") == kind)
+  ignore("taskLostBehavior = WAIT_FOREVER") {
+    Given("A resident app with 1 instance")
+    When("The task is lost")
+    Then("No timeout is scheduled") // can we easily verify this?
+    And("The task is not relaunched") // can we verify this without waiting?
   }
 
-  private[this] def create(app: AppDefinition) = {
-    val result = marathon.createAppV2(app)
-    result.code should be (201) //Created
-    extractDeploymentIds(result) should have size 1
+  ignore("relaunchEscalationTimeoutSeconds = 5s") {
+    Given("A resident app with 1 instance")
+    When("The task terminates")
+    And("We don't get an offer within the timeout")
+    Then("We launch a new task on any matching offer")
   }
 
-  private[this] def suspend(appId: PathId) = {
-    val updateResult = marathon.updateApp(appId, AppUpdate(instances = Some(0)), force = true)
-    updateResult.code shouldBe 200
-    waitForTasks(appId, 0) //make sure there are no more tasks
+  ignore("Persistent volumes are not destroyed when an app is destroyed") {
+    Given("a resident app with 1 running instance")
+    When("the app is deleted")
+    Then("the task gets killed")
+    And("the task is still stored")
+    When("the app is created again")
+    Then("the existing task will be picked up again")
+    And("the new task will use the existing persistent volume")
   }
 
-  class Fixture {
-    def appWithPersistentVolume(appId: PathId, containerPath: String, cmd: String): AppDefinition = {
+  object F {
```

Comment:
I guess your approach here is: since it has no state it can be object ?
I don't have a problem here, but I guess to make this a class called Fixture, that is created in the test case is a well known concept. And it allows to create state in that fixture.